diff --git framework/src/main/groovy/org/moqui/impl/entity/EntityDefinition.groovy framework/src/main/groovy/org/moqui/impl/entity/EntityDefinition.groovy
index 4fb38cd2..5e9fe0c8 100644
--- framework/src/main/groovy/org/moqui/impl/entity/EntityDefinition.groovy
+++ framework/src/main/groovy/org/moqui/impl/entity/EntityDefinition.groovy
@@ -221,6 +221,7 @@ class EntityDefinition {
             if (internalEntityNode.attribute("no-update-stamp") != "true") {
                 // automatically add the lastUpdatedStamp field
                 internalEntityNode.append("field", [name:"lastUpdatedStamp", type:"date-time"])
+                internalEntityNode.append("field", [name:"createdStamp", type:"date-time"])
             }
 
             ArrayList<MNode> fieldNodeList = internalEntityNode.children("field")
@@ -431,6 +432,7 @@ class EntityDefinition {
                 String aliasName = fi.name
                 // never auto-alias these
                 if ("lastUpdatedStamp".equals(aliasName)) continue
+                if ("createdStamp".equals(aliasName)) continue
                 // if specified as excluded, leave it out
                 ArrayList<MNode> excludeList = aliasAll.children("exclude")
                 int excludeListSize = excludeList.size()
diff --git framework/src/main/groovy/org/moqui/impl/entity/EntityJavaUtil.java framework/src/main/groovy/org/moqui/impl/entity/EntityJavaUtil.java
index 6029e72e..29078e5e 100644
--- framework/src/main/groovy/org/moqui/impl/entity/EntityJavaUtil.java
+++ framework/src/main/groovy/org/moqui/impl/entity/EntityJavaUtil.java
@@ -276,6 +276,7 @@ public class EntityJavaUtil {
 
         public final FieldInfo[] pkFieldInfoArray, nonPkFieldInfoArray, allFieldInfoArray;
         final FieldInfo lastUpdatedStampInfo;
+        final FieldInfo createdStampInfo;
         public final String allFieldsSqlSelect;
         final Map<String, String> pkFieldDefaults, nonPkFieldDefaults;
 
@@ -363,6 +364,7 @@ public class EntityJavaUtil {
             Map<String, String> pkFieldDefaultsTemp = new HashMap<>();
             Map<String, String> nonPkFieldDefaultsTemp = new HashMap<>();
             FieldInfo lastUpdatedTemp = null;
+            FieldInfo createdStampTemp = null;
             for (int i = 0; i < allFieldInfoSize; i++) {
                 FieldInfo fi = allFieldInfoList.get(i);
                 allFieldInfoArray[i] = fi;
@@ -386,6 +388,7 @@ public class EntityJavaUtil {
                     else nonPkFieldDefaultsTemp.put(fi.name, defaultStr);
                 }
                 if ("lastUpdatedStamp".equals(fi.name)) lastUpdatedTemp = fi;
+                if ("createdStamp".equals(fi.name)) createdStampTemp = fi;
             }
             createOnlyFields = createOnlyFieldsTemp;
             needsAuditLog = needsAuditLogTemp;
@@ -395,6 +398,7 @@ public class EntityJavaUtil {
             pkFieldDefaults = pkFieldDefaultsTemp.size() > 0 ? pkFieldDefaultsTemp : null;
             nonPkFieldDefaults = nonPkFieldDefaultsTemp.size() > 0 ? nonPkFieldDefaultsTemp : null;
             lastUpdatedStampInfo = lastUpdatedTemp;
+            createdStampInfo = createdStampTemp;
 
             pkFieldInfoArray = new FieldInfo[pkFieldInfoList.size()];
             pkFieldInfoList.toArray(pkFieldInfoArray);
diff --git framework/src/main/groovy/org/moqui/impl/entity/EntityValueBase.java framework/src/main/groovy/org/moqui/impl/entity/EntityValueBase.java
index 2c38f0d9..61aa08ba 100644
--- framework/src/main/groovy/org/moqui/impl/entity/EntityValueBase.java
+++ framework/src/main/groovy/org/moqui/impl/entity/EntityValueBase.java
@@ -982,6 +982,7 @@ public abstract class EntityValueBase implements EntityValue {
             for (String nonpkFieldName : this.getEntityDefinition().getNonPkFieldNames()) {
                 // skip the lastUpdatedStamp field
                 if ("lastUpdatedStamp".equals(nonpkFieldName)) continue;
+                if ("createdStamp".equals(nonpkFieldName)) continue;
 
                 final Object checkFieldValue = this.get(nonpkFieldName);
                 final Object dbFieldValue = dbValue.get(nonpkFieldName);
@@ -1035,6 +1036,7 @@ public abstract class EntityValueBase implements EntityValue {
             for (String nonpkFieldName : this.getEntityDefinition().getNonPkFieldNames()) {
                 // skip the lastUpdatedStamp field
                 if ("lastUpdatedStamp".equals(nonpkFieldName)) continue;
+                if ("createdStamp".equals(nonpkFieldName)) continue;
 
                 final Object checkFieldValue = this.get(nonpkFieldName);
                 final Object dbFieldValue = dbValue.get(nonpkFieldName);
@@ -1510,9 +1512,13 @@ public abstract class EntityValueBase implements EntityValue {
         final Long time = ecfi.transactionFacade.getCurrentTransactionStartTime();
         Long lastUpdatedLong = time != null && time > 0 ? time : System.currentTimeMillis();
         FieldInfo lastUpdatedStampInfo = ed.entityInfo.lastUpdatedStampInfo;
+        FieldInfo createdStampInfo = ed.entityInfo.createdStampInfo;
         if (lastUpdatedStampInfo != null && valueMapInternal.getByIString(lastUpdatedStampInfo.name, lastUpdatedStampInfo.index) == null)
             valueMapInternal.putByIString(lastUpdatedStampInfo.name, new Timestamp(lastUpdatedLong), lastUpdatedStampInfo.index);
 
+        if (createdStampInfo != null && valueMapInternal.getByIString(createdStampInfo.name, createdStampInfo.index) == null)
+            valueMapInternal.putByIString(createdStampInfo.name, new Timestamp(lastUpdatedLong), createdStampInfo.index);
+
         // do the artifact push/authz
         ArtifactExecutionInfoImpl aei = new ArtifactExecutionInfoImpl(entityName, ArtifactExecutionInfo.AT_ENTITY, ArtifactExecutionInfo.AUTHZA_CREATE, "create").setParameters(valueMapInternal);
         aefi.pushInternal(aei, !entityInfo.authorizeSkipCreate, false);
diff --git runtime/base-component/tools/screen/Tools/AutoScreen/AutoEdit/AutoEditDetail.xml runtime/base-component/tools/screen/Tools/AutoScreen/AutoEdit/AutoEditDetail.xml
index 3aecebf..5e45105 100644
--- runtime/base-component/tools/screen/Tools/AutoScreen/AutoEdit/AutoEditDetail.xml
+++ runtime/base-component/tools/screen/Tools/AutoScreen/AutoEdit/AutoEditDetail.xml
@@ -67,6 +67,7 @@ along with this software (see the LICENSE.md file). If not, see
                 <form-single name="CreateEntityValue" transition="create" dynamic="true">
                     <auto-fields-entity entity-name="${den}" field-type="edit"/>
                     <field name="aen"><default-field><hidden/></default-field></field>
+                    <field name="createdStamp"><default-field><hidden/></default-field></field>
                     <field name="den"><default-field><hidden/></default-field></field>
                     <field name="submitButton"><default-field title="Create"><submit/></default-field></field>
                 </form-single>
diff --git runtime/base-component/tools/screen/Tools/AutoScreen/AutoEdit/AutoEditMaster.xml runtime/base-component/tools/screen/Tools/AutoScreen/AutoEdit/AutoEditMaster.xml
index 723805c..bc1ae55 100644
--- runtime/base-component/tools/screen/Tools/AutoScreen/AutoEdit/AutoEditMaster.xml
+++ runtime/base-component/tools/screen/Tools/AutoScreen/AutoEdit/AutoEditMaster.xml
@@ -121,6 +121,8 @@ along with this software (see the LICENSE.md file). If not, see
             <!-- NOTE: lastUpdatedStamp hidden field added by auto-fields-entity include=nonpk so don't add another hidden field here -->
             <field name="lastUpdatedStamp"><default-field title="Last Updated">
                 <display format="yyyy-MM-dd HH:mm:ss.SSS z" also-hidden="false"/></default-field></field>
+            <field name="createdStamp"><default-field title="Created Updated">
+                <display format="yyyy-MM-dd HH:mm:ss.SSS z" also-hidden="false"/></default-field></field>
             <field name="submitButton"><default-field title="Update"><submit/></default-field></field>
         </form-single>
         <link url="getFormXml" text="Get Form XML" link-type="anchor"/>
diff --git runtime/base-component/tools/screen/Tools/AutoScreen/AutoFind.xml runtime/base-component/tools/screen/Tools/AutoScreen/AutoFind.xml
index 0d19667..cc0e0c4 100644
--- runtime/base-component/tools/screen/Tools/AutoScreen/AutoFind.xml
+++ runtime/base-component/tools/screen/Tools/AutoScreen/AutoFind.xml
@@ -65,6 +65,7 @@ along with this software (see the LICENSE.md file). If not, see
             <container-dialog id="CreateValueDialog" button-text="New Value">
                 <form-single name="CreateEntityValue" transition="create" dynamic="true">
                     <auto-fields-entity entity-name="${aen}" field-type="edit"/>
+                    <field name="createdStamp"><default-field><hidden/></default-field></field>
                     <field name="aen"><default-field><hidden/></default-field></field>
                     <field name="submitButton"><default-field title="Create"><submit/></default-field></field>
                 </form-single>