diff --git framework/src/main/groovy/org/moqui/impl/entity/FieldInfo.java framework/src/main/groovy/org/moqui/impl/entity/FieldInfo.java
index 52919f8c..cf83bf35 100644
--- framework/src/main/groovy/org/moqui/impl/entity/FieldInfo.java
+++ framework/src/main/groovy/org/moqui/impl/entity/FieldInfo.java
@@ -13,6 +13,7 @@
  */
 package org.moqui.impl.entity;
 
+import co.hotwax.util.OmsEntityCrypto;
 import org.moqui.BaseArtifactException;
 import org.moqui.entity.EntityException;
 import org.moqui.impl.context.L10nFacadeImpl;
@@ -397,7 +398,11 @@ public class FieldInfo {
             if (typeValue != 1) throw new EntityException("The encrypt attribute was set to true on non-String field " + name + " of entity " + entityName);
             String original = value.toString();
             try {
-                value = EntityJavaUtil.enDeCrypt(original, false, efi);
+                try {
+                    value = EntityJavaUtil.enDeCrypt(original, false, efi);
+                } catch (Exception e) {
+                    value = OmsEntityCrypto.decrypt(efi.ecfi.getEci(), this.ed.getEntityName(), OmsEntityCrypto.EncryptMethod.TRUE, original);
+                }
             } catch (Exception e) {
                 logger.error("Error decrypting field [" + name + "] of entity [" + entityName + "]", e);
                 // NOTE DEJ 20200310 instead of using encrypted value return very clear fake placeholder; this is a bad design
